src/lib/mpstream/mpstream.c

```diff
@@ -29,7 +29,9 @@ mpstream_panic_cb(void *error_ctx)
 void
 mpstream_reserve_slow(struct mpstream *stream, size_t size)
 {
-	stream->alloc(stream->ctx, stream->pos - stream->buf);
+	size_t alloc = stream->pos - stream->buf;
+	if (alloc > 0)
+		stream->alloc(stream->ctx, alloc);
```

Comment:
Nit: No need in the new local variable here:

```C
if (stream->pos != stream->buf)
        stream->alloc(stream->ctx, stream->pos - stream->buf);
```

Reply:
Done.
```diff
@@ -29,9 +29,8 @@ mpstream_panic_cb(void *error_ctx)
 void
 mpstream_reserve_slow(struct mpstream *stream, size_t size)
 {
-       size_t alloc = stream->pos - stream->buf;
-       if (alloc > 0)
-               stream->alloc(stream->ctx, alloc);
+       if (stream->pos != stream->buf)
+               stream->alloc(stream->ctx, stream->pos - stream->buf);
        stream->buf = stream->reserve(stream->ctx, &size);
        if (stream->buf == NULL) {
                diag_set(OutOfMemory, size, "mpstream", "reserve");
```